src/validation.cpp

```diff
@@ -210,9 +210,17 @@ class CChainState {
     bool RollforwardBlock(const CBlockIndex* pindex, CCoinsViewCache& inputs, const CChainParams& params) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 } g_chainstate;
 
-
-
-CCriticalSection cs_main;
+/**
+ * Mutex to guard access to validation specific variables, such as reading
+ * or changing the chainstate.
+ *
+ * This must also be locked when updating the transaction pool, e.g. on
```

Comment:
In commit "doc: Add comment to cs_main and mempool::cs" (fa64bb42e97b6c6c58e83badca88d386f99dbcd2)

I would say "may also need to be locked" instead of "must also be locked", and add "see CTxMemPool::cs comment for details."

Reply:
Done